bin/pyenv-installer

```diff
@@ -50,15 +52,78 @@ checkout "${GITHUB}/pyenv/pyenv-update.git"     "${PYENV_ROOT}/plugins/pyenv-upd
 checkout "${GITHUB}/pyenv/pyenv-virtualenv.git" "${PYENV_ROOT}/plugins/pyenv-virtualenv"  "master"
 checkout "${GITHUB}/pyenv/pyenv-which-ext.git"  "${PYENV_ROOT}/plugins/pyenv-which-ext"   "master"
 
-if ! command -v pyenv 1>/dev/null; then
-  { echo
-    colorize 1 "WARNING"
-    echo ": seems you still have not added 'pyenv' to the load path."
-    echo
+write_source() {
+  {
+    echo "Appending the following lines to $1:"
+    echo '  # pyenv'
+    echo '  export PATH="'"$PYENV_ROOT/bin:"'$PATH"'
+    echo '  eval "$(pyenv init --path)"'
+    echo '  eval "$(pyenv virtualenv-init -)"'
+    echo ''
   } >&2
 
-  { # Without args, `init` commands print installation help
-    "${PYENV_ROOT}/bin/pyenv" init || true
-    "${PYENV_ROOT}/bin/pyenv" virtualenv-init || true
-  } >&2
+  echo "" >>$1
+  echo '# pyenv' >>$1
+  echo 'export PATH="'"$PYENV_ROOT/bin:"'$PATH"' >>$1
+  echo 'eval "$(pyenv init --path)"' >>$1
+  echo 'eval "$(pyenv virtualenv-init -)"' >>$1
+
+  SRC_WRITTEN=true
+}
+
+add_userpath() {
+  OS="$(uname -s)"
+  CURRENT_SHELL="$(basename "$SHELL")"
+
+  if [[ "$OS" = "Darwin" ]]; then
+    if [[ "$CURRENT_SHELL" = "zsh" ]]; then
+      write_source "$HOME/.zprofile"
+    elif [[ "$CURRENT_SHELL" = "bash" ]]; then
+      write_source "$HOME/.bash_profile"
+    fi
+
+  elif [[ "$CURRENT_SHELL" = "zsh" ]]; then
+    # write to first match: zshrc/zprofile
+    if [[ -f $HOME/.zshrc ]]; then
+      write_source "$HOME/.zshrc"
+    elif [[ -f $HOME/.zprofile ]]; then
+      write_source "$HOME/.zprofile"
+    fi
+
+  elif [[ "$CURRENT_SHELL" = "bash" ]]; then
+    # bashrc
+    write_source "$HOME/.bashrc"
+
+    # write to first match: profile/bash_profile
+    if [[ -f $HOME/.profile ]]; then
+      write_source "$HOME/.profile"
+    elif [[ -f $HOME/.bash_profile ]]; then
+      write_source "$HOME/.bash_profile"
+    fi
+
+  else
+    {
+      echo "Add userpath for $CURRENT_SHELL is not currently supported"
+      echo "Please set up manually"
+    } >&2
+  fi
+}
+
+if ! command -v pyenv 1>/dev/null; then
```

Comment:
What's the logic behind this criterion?

Maybe rather check that shell config files don't have any mention of "pyenv"? Or run a new shell with `-i` and/or `--login` and check if anything related to Pyenv is set?

Reply:
I lost the context of this comment.

Could you check if this still being an issue after reviewing the updated code?
Thanks!

Reply:
AFAICS, this line is supposed to detect whether we need to add shell configuration stuff, or if it's already present.

I can't see how this test achieves that.

Since there's an infinite variety of setups, perhaps the best way of action is to see how other language managers do this -- to see what measures have proven to be sufficient in practice.

Reply:
This criterion is based on assumptions of how `pyenv-installer` might be used: 

1. Most users would run `pyenv-installer` on a newly configured system (without pyenv installed). 
2. The installer will update shell configuration stuff correctly, therefore `pyenv` would be detected after restarting the shell.

This criterion is not water-proof, especially for users that rerun the installer many times without restarting their shell. But it's probably the simplest way AFAICS.